content/articles/pointers-in-c#/index.md

```diff
@@ -0,0 +1,277 @@
+### Introduction
+
+In C#, data types are categorized into three categories: `pointer types`, `value types`, and `reference types`. The memory address of another kind is stored in a pointer-type variable. Since it lacks an individual variable, the reference data type holds a reference to another data type. Data types are explicitly included within the value data form. Integers, characters, and floating-point numbers, as well as the alphabet and numbers, are examples of value data types.
+
+### Table of contents
+
+1. [Making a pointer style declaration](#making-a-pointer-style-declaration)
+2. [Safe and unsafe codes](#safe-and-unsafe-codes)
+3. [Pinning an object](#Pinning-an-object)
+4. [Pointers and methods](#pointers-and-methods)
+5. [Conversions and pointers](#conversions-and-pointers)
+6. [Pointers and arrays](#pointers-and-arrays)
+7. [Pointers and structure](#pointers-and-structure)
+
+### Making a pointer style declaration
+In C#, we declare pointers as illustrated below:
+
+```c#
+type *variable_name;
+```
+
+Where `*` is called the de-reference administrator. The de-reference administrator or de-reference operator is used for getting the value from the address that the pointer refers to.
+
+Consider the following example: 
+
+```c#
+int w=76
+int *ptr =&w;
+Console.WriteLine((int)ptr) 
+Console.WriteLine(*ptr)     .
+```
+
+`w` is a pointer variable that can hold the position of an `int` sort. The operator `&` is known as a reference operator, and it is used to get a variable address. The memory address of the variable `w`, which can be assigned to a pointer variable, is specified by the symbols `&w`.
+
+### Safe and unsafe codes
+**Safe codes** are C# keywords that run under the `Common Language Runtime's supervision (CLR)` while **Unsafe codes** are C# keywords that execute outside the management of the CLR. Unlike C++ and C programming languages, which use safe codes with pointers, the C# programming language only allows the use of unsafe codes. The unsafe codes may be used as a modifier or to label a group of statements as unsafe. Common language Runtime translates safe codes into software instructions, which are then executed by the computer's CPU.
+
+The example below uses the unsafe codes:
+
+```C#
+using System;
+namespace UnsafeCodeApplication
+
+{
+    class Demo
+    {
+        public void Method()
+        {
+            unsafe
+            {
+                int m = 60;
+                int n = 30;
+                int* ptr1 = &m;
+                int* ptr2 = &n;
+                Console.WriteLine(*ptr1);    
+                Console.WriteLine(*ptr2);    
+                Console.WriteLine((int)ptr1); 
+                Console.WriteLine((int)ptr2); 
+            }
+        }
+    }
+    class Example
+    {
+        public static void Main()
+        {
+            Demo d = new Demo();
+            d.Method();
+        }
+    }
+} 
+```
+
+And the output is:
+
+```
+60
+30
+1605887284
+1605887280
```

Comment:
The reader should know that the last two results may be different in his machine based on the values addresses.